substrate-node/pallets/pallet-tft-bridge/src/migrations/v2.rs

```diff
@@ -17,9 +17,10 @@ impl<T: Config> OnRuntimeUpgrade for MigrateBurnTransactionsV2<T> {
     #[cfg(feature = "try-runtime")]
     fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
         info!("current pallet version: {:?}", PalletVersion::<T>::get());
-        if PalletVersion::<T>::get() != types::StorageVersion::V1 {
-            return Ok(Vec::<u8>::new());
-        }
+        ensure!(
```

Comment:
The old check allowed early returns and made sure idempotency checks passed.
Can we ensure the migration remains idempotent?

Reply:
I ll update